aws/src/main/java/org/apache/iceberg/aws/kms/KmsClientProperties.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.aws.kms;
+
+import java.io.Serializable;
+import java.util.Map;
+import org.apache.iceberg.util.PropertyUtil;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.core.retry.RetryMode;
+import software.amazon.awssdk.services.kms.KmsClientBuilder;
+
+public class KmsClientProperties implements Serializable {
+
+  /** This property is used to configure {@link RetryMode} for AWS KMS client */
+  public static final String KMS_RETRY_MODE = "kms.retry.mode";
+
+  public static final String KMS_RETRY_MODE_DEFAULT = "LEGACY";
+
+  private RetryMode kmsRetryMode;
+
+  public KmsClientProperties() {
+    this.kmsRetryMode = RetryMode.valueOf(KMS_RETRY_MODE_DEFAULT);
+  }
+
+  public KmsClientProperties(Map<String, String> properties) {
```

Comment:
nit: I would add the link to AWS configBuilder configuration, especially about retry mode (to know the accepted values for retry mode).